python/pyarrow/parquet.py

```diff
@@ -877,17 +894,22 @@ def filter_accepts_partition(self, part_key, filter, level):
 
         f_type = type(f_value)
 
-        if isinstance(f_value, set):
+        if op in {'in', 'not in'}:
+            if not (isinstance(f_value, Container) and
+                    isinstance(f_value, Iterable)):
```

Comment:
Could maybe also check for `Collection` instead of the combination of Container and Iterable?

Reply:
Good idea.

Reply:
Hi Joris. I changed to use Collection. But the build failed for python 3.5 which does not support Collection.
What is your take on this?
Thanks.

Reply:
Collection is not supported in python 3.5, so roll back to Container and Iterable.